### PR TITLE
:bug: return None when rounding 0

### LINKED ISSUE #171: Error with graph whan a column has a boolean with always same value
Hi,
we encounter an error when trying to draw a plot with a boolean column whith always the same value (True or False).
It appears when showing the mean aggregation size.
When doing this, this line (https://github.com/predict-idlab/plotly-resampler/blob/39072f55409a12c7955cd977659c048cb28865c2/plotly_resampler/figure_resampler/figure_resampler_interface.py#L307) always return 0, because you use np.diff
By consequence, this following line (https://github.com/predict-idlab/plotly-resampler/blob/39072f55409a12c7955cd977659c048cb28865c2/plotly_resampler/figure_resampler/utils.py#L171) is doing a math.log10 on 0, and we encounter a math domain error (because log(0) is impossible).

We are bypassing this by desactivating the show of mean aggregation size (for all our graphes), but we love this feature, so if you could find a workaround it would be great! Just let me know, so I can tell to my users if this number will come back or no. And thanks for your great work!

### PR DESCRIPTION
**Pull request by jvdd**
Fixes #171 

When rounding numbers or timedelta's, None is returned when the number (i.e., diff) == 0

- [x] update code to make timedelta & float rounding return None when rounding 0
- [x] update docs
- [x] update tests

### CODE REVIEW

**Comment by jvdd** [comment]
Is ready for review! @jonasvdd

File: tests/test_utils.py [hunk 1]
```diff
@@ -109,3 +110,4 @@ def test_round_int_str():
     assert round_number_str(950_001) == "1M"
     assert round_number_str(1_950_001) == "2M"
     assert round_number_str(111_950_001) == "112M"
+    assert round_number_str(0) is None
```
Comment (jonasvdd) [inline tests/test_utils.py:113]: maybe also test negative numbers?

**Review by jonasvdd — COMMENTED** [review]
Good catch! 🔍 

I would just add two little tests for the negative time and number check and then it's mergeable!

**Comment by jvdd** [comment]
Negative numbers should now be supported in `round_number_str`

**Review by jonasvdd — APPROVED** [review]
LGTM!
